pynubank/nubank.py

```diff
@@ -49,19 +50,49 @@ def _make_graphql_request(self, graphql_object):
         return json.loads(request.content.decode('utf-8'))
 
     def authenticate(self, cpf, password):
-        body = {
-            "grant_type": "password",
-            "login": cpf,
-            "password": password,
-            "client_id": "other.conta",
-            "client_secret": "yQPeLzoHuJzlMMSAjC-LgNUJdUecx8XO"
-        }
-        request = requests.post(self.TOKEN_URL, json=body, headers=self.headers)
-        if request.status_code != 200:
-            message = '{} ({})'.format(request.reason, request.status_code)
-            raise NuException('Authentication failed. {}'.format(message))
 
-        data = json.loads(request.content.decode('utf-8'))
+        cache_path = os.path.join(self.CACHE_DIR, 'auth.json')
+        os.makedirs(os.path.dirname(cache_path), exist_ok=True)
+        token_valid = False
+
+        if os.path.isfile(cache_path):
+
+            try:
+                with open(cache_path, 'r+') as file:
+                    data = json.loads(file.read())
+
+            except FileNotFoundError as exc:
+                print(exc)
```

Comment:
I think that using `print` is not a good idea. I wouldn't want to have this exceptions printed when I am running pynubank. I think using logger or just not print is a better approach.

Reply:
Was using for debuging purposes and forgot to remove it, my bad. I'm also in favor of using `logging.exception()` instead of printing.
Catching these exceptions just sets the validity of the token anyway, if there is something wrong with it we should simply generate a new one.